install/upgrade_to_2.13.php

```diff
@@ -140,4 +141,165 @@ protected function addDashboardVisibility() {
+         ],
+         'glpi_plugin_formcreator_sections' => [
+            'id',
+            'plugin_formcreator_forms_id',
+         ],
+         'glpi_plugin_formcreator_targetchanges' => [
+            'id',
+            'plugin_formcreator_forms_id',
+            'changetemplates_id',
+            'due_date_question',
+            'urgency_question',
+            'destination_entity_value',
+            'category_question',
+            'sla_question_tto',
+            'sla_question_ttr',
+            'ola_question_tto',
+            'ola_question_ttr',
+         ],
+         'glpi_plugin_formcreator_targettickets' => [
+            'id',
+            'plugin_formcreator_forms_id',
+            'type_question',
+            'tickettemplates_id',
+            'due_date_question',
+            'urgency_question',
+            'destination_entity_value',
+            'category_question',
+            'associate_question',
+            'location_question',
+            'sla_question_tto',
+            'sla_question_ttr',
+            'ola_question_tto',
+            'ola_question_ttr',
+         ],
+         'glpi_plugin_formcreator_targetproblems' => [
+            'id',
+            'plugin_formcreator_forms_id',
+            'problemtemplates_id',
+            'urgency_question',
+            'destination_entity_value',
+            'category_question',
+         ],
+         'glpi_plugin_formcreator_targets_actors' => [
+            'id',
+            'items_id',
+            'actor_value',
+         ],
+         'glpi_plugin_formcreator_issues' => [
+            'id',
+            'items_id',
+            'entities_id',
+            'requester_id',
+            'users_id_validator',
+            'groups_id_validator',
+            'users_id_recipient',
+         ],
+         'glpi_plugin_formcreator_items_targettickets' => [
+            'id',
+            'plugin_formcreator_targettickets_id',
+            'items_id',
+         ],
+         'glpi_plugin_formcreator_questiondependencies' => [
+            'id',
+            'plugin_formcreator_questions_id',
+            'plugin_formcreator_questions_id_2',
+         ],
+         'glpi_plugin_formcreator_questionregexes' => [
+            'id',
+            'plugin_formcreator_questions_id',
+         ],
+         'glpi_plugin_formcreator_questionranges' => [
+            'id',
+            'plugin_formcreator_questions_id',
+         ],
+         'glpi_plugin_formcreator_forms_languages' => [
+            'id',
+            'plugin_formcreator_forms_id',
+         ],
+      ];
+
+      foreach ($tables as $table => $fields) {
+         foreach ($fields as $field) {
+            if ($field == 'id') {
+               $type = 'autoincrement';
+            } else {
+               $type = "INT " . DBConnection::getDefaultPrimaryKeySignOption() . " NOT NULL DEFAULT 0";
+            }
+            $this->migration->changeField($table, $field, $field, $type);
+         }
+      }
```

Comment:
This migration is already done by GLPI, you do not need to do it.

Reply:
There is at least 1 columns having a name which does not match the FK manimg pattern (it may contain  FK of different itemtypes). Then this code is required. 

Also, I update all columns mostly to pass the unit tests of the plugin (upgrade from 2.5.0). It is designed to build a GLPI instance then add tables of an old version of Formcreator. In such scenario, the columns are not upgraded by GLPI and when it is time to compare fresh tables against upgraded ones, the difference will turn the tests red.

Reply:
I updated unit tests to rely when possible on GLPI for this migration. But several columns don't upgrade because they match the expected type, but don't have a devault value, so this code shall remain.